Sample.BasicComConnector/NewSimClient.cs

```diff
@@ -36,27 +38,163 @@ public Task TestConnection(CancellationToken _token)
 
     protected override void PreShutdown()
     {
+        CleanupExtractedDirectories();
```

Comment:
Please don't do cleanup of extracted directories here, do it at the end of the simulation run